common/lib/xmodule/xmodule/tests/test_conditional.py

```diff
@@ -8,7 +8,8 @@
 from xblock.field_data import DictFieldData
 from xblock.fields import ScopeIds
 from xmodule.error_module import NonStaffErrorDescriptor
-from opaque_keys.edx.locations import SlashSeparatedCourseKey, Location
+from opaque_keys.edx.keys import CourseKey
```

Comment:
This test was causing me some grief, and I thought it was related to the change to CourseKey.from_string. But I finally figured out that the issue exists on master as well (I just hadn't tried running the test in the particular way that exhibits the issue until after I had made the CourseKey change).

If you run this test (in its larger suite) via "paver test_lib common/lib/xmodule" (which runs the nosetests command), then it passes on devstack. It also passes if you run it under CMS' settings via "cms --settings test test common/lib/xmodule/xmodule/tests/test_conditional.py". A few testpoints fail though when run under LMS' settings because in the LMS environment, edxnotes is installed as a mixin that attaches to the HTMLModule. The edxnotes code then tries to access self.descriptor.runtime.modulestore, which fails because there is no modulestore (this test runs with an XMLStore). But as previously mentioned, this behavior is seen on master as well.